tests/units/offset/test_total_claimable.cairo

```diff
@@ -17,33 +17,37 @@ func __setup__{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
 }
 
 @external
-func test_balance_of{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() {
+func test_total_claimable{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() {
     alloc_locals;
 
     // prepare farmer instance
     let (local context) = prepare();
-    let zero = Uint256(low=0, high=0);
     let one = Uint256(low=1, high=0);
-    let ten = Uint256(low=10, high=0);
-    let hundred = Uint256(low=100, high=0);
     let (contract_address) = get_contract_address();
 
+    %{ mock_call(context.mocks.carbonable_project_address, "isSetup", [1]) %}
     %{ mock_call(context.mocks.carbonable_project_address, "transferFrom", [1]) %}
-    %{ mock_call(context.mocks.carbonable_project_address, "ownerOf", [ids.contract_address]) %}
-    %{ mock_call(context.mocks.carbonable_project_address, "balanceOf", [1, 0]) %}
     %{ mock_call(context.mocks.carbonable_project_address, "totalSupply", [1, 0]) %}
     %{ mock_call(context.mocks.carbonable_project_address, "tokenByIndex", [1, 0]) %}
+    %{ mock_call(context.mocks.carbonable_project_address, "ownerOf", [ids.contract_address]) %}
+    %{ mock_call(context.mocks.carbonable_project_address, "getAbsorption", [1000000]) %}
+    %{ mock_call(context.mocks.carbonable_project_address, "getCurrentAbsorption", [3000000]) %}
 
+    // Anyone
     %{ stop=start_prank(context.signers.anyone) %}
+
+    // Total claimable is 0
+    let (total_claimable) = CarbonableOffseter.total_claimable();
+    assert total_claimable = 0;
+
+    // Deposit token #1
     let (success) = CarbonableOffseter.deposit(token_id=one);
     assert success = 1;
-    %{ stop() %}
-
-    let (balance) = CarbonableOffseter.balance_of(address=context.signers.anyone);
-    assert balance = 1;
 
-    let (balance) = CarbonableOffseter.balance_of(address=context.signers.admin);
-    assert balance = 0;
+    // Total claimable is 3 - 1 = 2
```

Comment:
if the user never claimed and have deposit his token from the beginning, it should be have  `total_claimable() = 3000000` ?

Reply:
The trick occurs with the following pranks:

```
%{ mock_call(context.mocks.carbonable_project_address, "getAbsorption", [1000000]) %}
%{ mock_call(context.mocks.carbonable_project_address, "getCurrentAbsorption", [3000000]) %}
```
- `getAbsorption` is computed to get the absorption at the deposit time
- `getCurrentAbsorption` is computed to get the current absorption

The claimable absorption is the difference between both.
The purpose of the test was to ensure that the difference is done correctly whenever the deposit and computation are done.